### DISCUSSION TITLE
Transmitted Bytes and processOutgoing

### DISCUSSION THREAD

**Discussion by faragher** [markqvist/Reticulum#286] [Help & Questions] [answered]
processOutgoing, in the RNode Interface, currently adds the initial length of the data to the TXB, as opposed to the escaped length. Is this intended?

**Comment by markqvist** [comment, marked as answer]
Hmm, yeah, it is actually intended, since that is the actual amount of *raw data* transmitted, even though it does not necesarrily correspond to the exact amount of bits ocurring over the air.

The data is only escaped while travelling over the USB/BT/Serial connection. When the RNode receives it, it is unescaped and kept in RAM as it's actual data representation, until it is sent out over the air, where again certain bytes or sequences may be escaped (which is handled by the LoRa modem though).

So while the particulars can be a little confusing, I think this is the right way to do it.

**Reply by faragher** [reply]
Of course. It didn't occur to me that the transmission itself wasn't escaped. Got it, thanks.
